public/app/core/components/TraceToLogs/TraceToLogsSettings.tsx

```diff
@@ -54,19 +58,60 @@ export function TraceToLogsSettings({ options, onOptionsChange }: Props) {
         </InlineField>
       </InlineFieldRow>
 
+      {options.jsonData.tracesToLogs?.mapTagNamesEnabled ? (
+        <InlineFieldRow>
+          <InlineField
+            tooltip="Tags that will be used in the Loki query. Default tags: 'cluster', 'hostname', 'namespace', 'pod'"
+            label="Tags"
+            labelWidth={26}
+          >
+            <KeyValueInput
+              keyPlaceholder="Tag"
+              values={options.jsonData.tracesToLogs?.mappedTags ?? []}
+              onChange={(v) =>
+                updateDatasourcePluginJsonDataOption({ onOptionsChange, options }, 'tracesToLogs', {
+                  ...options.jsonData.tracesToLogs,
+                  mappedTags: v,
+                })
+              }
+            />
+          </InlineField>
+        </InlineFieldRow>
+      ) : (
+        <InlineFieldRow>
+          <InlineField
+            tooltip="Tags that will be used in the Loki query. Default tags: 'cluster', 'hostname', 'namespace', 'pod'"
+            label="Tags"
+            labelWidth={26}
+          >
+            <TagsInput
+              tags={options.jsonData.tracesToLogs?.tags}
+              width={40}
+              onChange={(tags) =>
+                updateDatasourcePluginJsonDataOption({ onOptionsChange, options }, 'tracesToLogs', {
+                  datasourceUid: options.jsonData.tracesToLogs?.datasourceUid,
```

Comment:
`...options.jsonData.tracesToLogs` is missing here. Changing tags clears all other settings.

Reply:
Good catch